src/main/java/com/spotify/dns/ErrorHandler.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2015 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.spotify.dns;
+
+/**
+ * Error handler callback for errors thrown by the {@link DnsSrvResolver} used in a
+ * {@link DnsSrvWatcher}.
+ */
+public interface ErrorHandler {
+
+  /**
+   * Handles a {@link DnsException} for the given FQDN.
+   *
+   * @param fqdn       The FQDN that was resolved
+   * @param exception  The exception thrown
+   */
+  void handle(String fqdn, DnsException exception);
```

Comment:
What's the rationale for only allowing clients to handle DnsExceptions as opposed to any exception?